### PR TITLE
Add validation of plugin's configuration before .save()

### PR DESCRIPTION
**Pull request by korycins**
Add new method to `BasePlugin` - `validate_plugin_configuration`. Manager will run it before saving new changes. If validation fails, it will raise `ValidationError` and will return error_msg over API.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for 6a243fac91519b0cb7601f80587d75c8d8196a34 (korycins/saleor @ feature/plugin_manager_add_config_validation)
Base comparison is f01365e08d32ff9f1d7383736683983694909056.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  add billing address to checkout     	         34	         34	             20
  add shipping to checkout            	          7	          7	              0
  checkout payment charge             	         14	         14	              0
  complete checkout                   	          6	          6	              0
  create checkout                     	         48	         48	             24

# api.benchmark homepage
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve main menu                  	          5	          5	              0
  retrieve product list               	          4	          4	              0
  retrieve secondary menu             	          5	          5	              0
  retrieve shop                       	          2	          2	              0

# api.benchmark product
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  product details                     	         13	         13	              3

# api.benchmark variant
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve variant list               	         15	         15	              8
```

</p>
</details>

File: saleor/extensions/plugins/avatax/plugin.py [hunk 1]
```diff
@@ -418,6 +420,23 @@ def taxes_are_enabled(self, previous_value: bool) -> bool:
...
+        configuration = plugin_configuration.configuration
+        configuration = {item["name"]: item["value"] for item in configuration}
+        if not configuration["Username or account"]:
+            missing_fields.append("Username or account")
+        if not configuration["Password or license"]:
+            missing_fields.append("Password or license")
+
+        if plugin_configuration.active and missing_fields:
+            error_msg = (
+                "To enable a plugin you need to provide missing value for the fields: "
```
Comment (maarcingebala) [inline saleor/extensions/plugins/avatax/plugin.py]:
```suggestion
                "To enable a plugin, you need to provide values for the following fields: "
```

File: saleor/extensions/plugins/avatax/plugin.py [hunk 2]
```diff
@@ -418,6 +420,23 @@ def taxes_are_enabled(self, previous_value: bool) -> bool:
...
+            )
+            raise ValidationError({"active": error_msg + ", ".join(missing_fields)})
```
Comment (maarcingebala) [inline saleor/extensions/plugins/avatax/plugin.py]: I would raise this error as a general one, without assigning it to the `active` field. It's related to a few fields usually.

File: saleor/extensions/plugins/vatlayer/plugin.py [hunk 3]
```diff
@@ -320,6 +322,17 @@ def get_tax_rate_percentage_value(
         tax = taxes.get(rate_name) or taxes.get(DEFAULT_TAX_RATE_NAME)
         return Decimal(tax["value"])
 
+    @classmethod
+    def validate_plugin_configuration(cls, plugin_configuration: "PluginConfiguration"):
+        """Validate if provided configuration is correct."""
+        if not settings.VATLAYER_ACCESS_KEY and plugin_configuration.active:
+            raise ValidationError(
+                {
+                    "active": "Cannot be enabled without provided "
```
Comment (maarcingebala) [inline saleor/extensions/plugins/vatlayer/plugin.py]: Same here about assigning the error to the `active` field.

File: saleor/extensions/plugins/vatlayer/plugin.py [hunk 4]
```diff
@@ -320,6 +322,14 @@ def get_tax_rate_percentage_value(
         tax = taxes.get(rate_name) or taxes.get(DEFAULT_TAX_RATE_NAME)
         return Decimal(tax["value"])
 
+    @classmethod
+    def validate_plugin_configuration(cls, plugin_configuration: "PluginConfiguration"):
+        """Validate if provided configuration is correct."""
+        if not settings.VATLAYER_ACCESS_KEY and plugin_configuration.active:
+            raise ValidationError(
+                "Cannot be enabled without provided " "'settings.VATLAYER_ACCESS_KEY'"
```
Comment (maarcingebala) [inline saleor/extensions/plugins/vatlayer/plugin.py]:
```suggestion
                "Cannot be enabled without provided 'settings.VATLAYER_ACCESS_KEY'"
```

**Pull request merged**